### DISCUSSION TITLE
Update LCX through HA (run "update")

### DISCUSSION THREAD

**Discussion by walberjunior** [tteck/Proxmox#2793] [General]
Hello, I want to manage LXC updates through HA and run the "update" command, but it would have to be done without confirmation via the UI.
Any way to do this?

**Comment by LarsNorgaard** [comment]
You could run something like Olivetin and a command like this:
`pct exec 112 -- bash -c "apt-get update 2>/dev/null | grep 'packages.*upgraded'; apt list --upgradable && apt-get -yq dist-upgrade 2>&1"`

**Comment by walberjunior** [comment]
Running the command in HA is not a problem, I can execute the commands via ssh.

  The problem I encounter is that not all lxc that I use are updated with apt update/ apt upgrade, the Linux part is updated, but the main software is not.  

Cloudflare and adguard are some that I remember off the top of my head.

I don't understand much about Linux, but I'll test the command.

**Comment by asylumexp** [comment]
>  I want to manage LXC updates through HA and run the "update" command

Are you wanting to update all at once or just one button per lxc?

If you want to do all at once you could just modify the update lxc script in the misc dir to remove the confirmation prompt and the ui asking which lxcs to skip

**Comment by walberjunior** [comment]
I want to update individually.
Something like "update -y" would do the job, but in this case the -y option doesn't "exist".

This is an example of the shell command I currently use for some lxc, but as I said before, it doesn't work for all lxc:
for qbittorrent
```
'ssh -i /config/ssh_key/ssh_key -o StrictHostKeyChecking=no -q root@192.168.0.97 "apt update && apt upgrade -y"'
```
For Z2M there is a script built in /opt
```
ssh -i /config/ssh_key/ssh_key -o StrictHostKeyChecking=no -q root@192.168.0.98 "apt update && apt upgrade -y && /opt/zigbee2mqtt/./update.sh"
```
Where is this misc folder? I could not find it

**Reply by asylumexp** [reply]
https://github.com/tteck/Proxmox/tree/main/misc
